File: pset6/caesar/caesar.py
#caesar.c ported to python 3#
#############################

import sys
import cs50


def main(argv):
    # get key as second argument and convert to int
    if len(sys.argv) != 2:
        print("caesar.py <key>")
        sys.exit(1)

    key = int(sys.argv[1])

    # prompt until user gives a string
    while True:
        plntxt = cs50.get_string("plaintext: ")
        if plntxt != None:
            break
        print("Plaintext only")

    # for each char in plaintext, determine type, convert to number and calculate,
    # then convert back to char and print
    print("ciphertext: ", end='')
    result = []
    for c in plntxt:
        if c.isalpha and c.isupper():
            result = (ord(c) - 65 + key) % 26 + 65
        elif c.isalpha and c.islower():
            result = (ord(c) - 97 + key) % 26 + 97
        else:
            result = ord(c)
        print(chr(result), end='')
    print()


if __name__ == "__main__":
    main(sys.argv)

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

string plain;
int key;
int charref;

int main(int argc, string argv[])
{
    //if user does not provide a 2nd argument then return error
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    /* put second argument into key variable */
    key = atoi(argv[1]);
    
    /* prompt for plaintext and put into plain variable */
    printf("plaintext: ");
    plain = get_string();

    printf("ciphertext: ");
    for (int i = 0, j = strlen(plain); i < j; i++)
    {
    /* if char is alphabetical then convert to cipher and print */
        if (isalpha(plain[i]))
        {
            if (isupper(plain[i]))
            {
                printf("%c", (plain[i] - 'A' + key) % 26 + 'A');
            }
            if (islower(plain[i]))
            {
                printf("%c", (plain[i] - 'a' + key) % 26 + 'a');
            }
        }
    /* if char is not in the alphabet then do nothing and just print it*/
        else 
        {
            printf("%c", plain[i]);
        }
    }
printf("\n");
return 0;
}